### PR TITLE
FPLA 893: Solicitor email and name mandatory in application

### PR DESCRIPTION
**Pull request by lewisbirks**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-893

### Change description ###

Made the solicitor name and email mandatory through validation upon case submission.
Introduced the ability for an error message to contain multiple points spread over multiple lines.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseValidatorServiceTest.java [hunk 1]
```diff
@@ -174,13 +181,30 @@ void shouldNotReturnAnErrorWhenFirstRespondentHasFullNameButNotSecondRespondent(
...
+            .solicitor(solicitor())
             .respondents1(respondents())
             .build();
 
         List<String> errors = caseValidatorService.validateCaseDetails(caseData);
         assertThat(errors).isEmpty();
     }
 
+    @Test
+    void shouldReturnAnErrorWhenSolicitorDoesNotHaveNameAndEmail() {
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseValidatorServiceTest.java]: Do we need to test this separately? You've asserted this twice above no?

Comment (lewisbirks) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseValidatorServiceTest.java]: Good point will remove

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java [hunk 2]
```diff
@@ -58,6 +58,9 @@ private final Respondent getFirstRespondent() {
     }
 
     private final Proceeding proceeding;
+
+    @NotNull(message = "Enter the solicitor's full name" + VALIDATION_SEPARATOR + "Enter the solicitor's email")
```
Comment (swalker125) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java]: I think the case data level validation is normally "you need to add details for solicitor" and then the Solicitor level validation is a "you need to fill in these fields"

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java]: Usually yes but it wasn't mentioned on the ticket so I kept it consistent.

Comment (Jamesnelson117) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java]: Confirm with Mia, for consistency we should really have - 

In the applicant section :
- You need to add details to applicants
- You need to add details to solicitor (or similar)

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java]: After discussion with Mia, have made it consistent with the rest of the messages when the event hadn't been accessed

**Pull request merged**